src/app/shared/components/template/utils/template-utils.ts

```diff
@@ -1,27 +1,35 @@
 import { FlowTypes } from "src/app/shared/model";
 import { arrayToHashmap } from "src/app/shared/utils";
 
-/**
```

Comment:
These are the modified methods that perform template deep merging, the changes are not intuitive as the 2 main methods were essentially flipped to allow merge with the template row processing methods (per TODO item below)